packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts

```diff
@@ -46,41 +46,72 @@ import {
+	 * However, fuzz schemas always have the same field names, so schema-dependent
+	 * APIs such as the tree reading API will work correctly anyway.
+	 *
+	 * TODO: The schema for each client should be properly updated if "afterSchemaChange" (or equivalent event) occurs
+	 * once schema ops are supported.
+	 */
+	currentSchema: FuzzNodeSchema;
+};
+
+export type FuzzTransactionView = ITreeViewFork<typeof fuzzSchema.rootFieldSchema> & {
+	/**
+	 * This client's current stored schema, which dictates allowable edits that the client may perform.
+	 * @remarks - The type of this field isn't totally correct, since the supported schema for fuzz nodes changes
+	 * at runtime to support different primitives (this allows fuzz testing of schema changes).
+	 * However, fuzz schemas always have the same field names, so schema-dependent
+	 * APIs such as the tree reading API will work correctly anyway.
+	 *
+	 * TODO: The schema for each client should be properly updated if "afterSchemaChange" (or equivalent event) occurs
+	 * once schema ops are supported.
+	 */
+	currentSchema: FuzzNodeSchema;
+};
 
 export interface FuzzTestState extends DDSFuzzTestState<SharedTreeFactory> {
 	/**
-	 * Schematized view of clients. Created lazily by viewFromState.
+	 * Schematized view of clients and their nodeSchemas. Created lazily by viewFromState.
 	 *
 	 * SharedTrees undergoing a transaction will have a forked view in {@link transactionViews} instead,
 	 * which should be used in place of this view until the transaction is complete.
 	 */
-	view?: Map<ISharedTree, FlexTreeView<typeof fuzzSchema.rootFieldSchema>>;
+	view?: Map<ISharedTree, FuzzView>;
 	/**
-	 * Schematized view of clients undergoing transactions.
+	 * Schematized view of clients undergoing transactions with their nodeSchemas.
 	 * Edits to this view are not visible to other clients until the transaction is closed.
 	 *
 	 * Maintaining a separate view here is necessary since async transactions are not supported on the root checkout,
 	 * and the fuzz testing model only simulates async transactions.
 	 */
-	transactionViews?: Map<ISharedTree, ITreeViewFork<typeof fuzzSchema.rootFieldSchema>>;
+	transactionViews?: Map<ISharedTree, FuzzTransactionView>;
 }
 
 export function viewFromState(
 	state: FuzzTestState,
 	client: Client<SharedTreeFactory> = state.client,
 	initialTree: TreeContent<typeof fuzzSchema.rootFieldSchema>["initialTree"] = undefined,
-): FlexTreeView<typeof fuzzSchema.rootFieldSchema> {
+): FuzzView {
 	state.view ??= new Map();
+
 	return (
 		state.transactionViews?.get(client.channel) ??
-		getOrCreate(state.view, client.channel, (tree) =>
-			tree.schematizeInternal({
+		getOrCreate(state.view, client.channel, (tree) => {
+			const fuzzView = tree.schematizeInternal({
 				initialTree,
 				schema: fuzzSchema,
 				allowedSchemaModifications: AllowedUpdateType.None,
-			}),
-		)
+			}) as FuzzView;
+			fuzzView.currentSchema = fuzzNode;
```

Comment:
Before assigning, we should

```typescript
// This means the tree view implementation already uses a property with the name `currentSchema` and the fuzz harness should pick a different name.
assert.equal(fuzzView.currentSchema, undefined)
```

Reply:
this ensures if we update prod codepath to use this name (which is *kinda* plausible), we won't get confusing errors down the line.